examples/plot_hmm_sampling_and_decoding.py

```diff
@@ -0,0 +1,115 @@
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from hmmlearn import hmm
+from sklearn.model_selection import RepeatedKFold
+
+# Prepare parameters for a 4-components HMM
+# Initial population probability
+startprob = np.array([0.6, 0.3, 0.1, 0.0])
+# The transition matrix, note that there are no transitions possible
+# between component 1 and 3
+transmat = np.array([[0.7, 0.2, 0.0, 0.1],
+                     [0.3, 0.5, 0.2, 0.0],
+                     [0.0, 0.3, 0.5, 0.2],
+                     [0.2, 0.0, 0.2, 0.6]])
+# The means of each component
+means = np.array([[0.0, 0.0],
+                  [0.0, 11.0],
+                  [9.0, 10.0],
+                  [11.0, -1.0]])
+# The covariance of each component
+covars = .5 * np.tile(np.identity(2), (4, 1, 1))
+
+# Build an HMM instance and set parameters
+gen_model = hmm.GaussianHMM(n_components=4, covariance_type="full")
+
+# Instead of fitting it from the data, we directly set the estimated
+# parameters, the means and covariance of the components
+gen_model.startprob_ = startprob
+gen_model.transmat_ = transmat
+gen_model.means_ = means
+gen_model.covars_ = covars
+
+# Generate samples
+X, Z = gen_model.sample(500)
+
+# Plot the sampled data
+plt.plot(X[:, 0], X[:, 1], ".-", label="observations", ms=6,
+         mfc="orange", alpha=0.7)
+
+# Indicate the component numbers
+for i, m in enumerate(means):
+    plt.text(m[0], m[1], 'Component %i' % (i + 1),
+             size=17, horizontalalignment='center',
+             bbox=dict(alpha=.7, facecolor='w'))
+plt.legend(loc='best')
+plt.show()
+
+# %%
+# Now, let's ensure we can recover our parameters.
+
+scores = list()
+models = list()
+rkf = RepeatedKFold(n_splits=2, n_repeats=10, random_state=99)
+for n_components in (3, 4, 5):
+    for train_idx, validate_idx in rkf.split(X):
+        # define our hidden Markov model
+        model = hmm.GaussianHMM(n_components=n_components,
+                                covariance_type='full',
+                                random_state=len(models), n_iter=10)
+        try:
+            model.fit(X[train_idx])
+            models.append(model)
+            scores.append(model.score(X[validate_idx]))
+            print(f'Converged: {model.monitor_.converged}\tScore:{scores[-1]}')
+        except ValueError:
+            pass  # sometimes there is a linear algebra error
+
+# get the best model
+model = models[np.argmax(scores)]
+n_states = model.n_components
+print(f'The best model had a score of {max(scores)} and {n_states} '
+      'states')
+
+# use the model to predict states
+states = model.predict(X)
+
+# %%
+# Let's plot our states compared to those generated and our transition matrix
+# to get a sense of our model. We can see that the recovered states follow
+# the same path as the generated states, just with the identities of the
+# states transposed. The same is true for the transition matrix.
+
+# plot model states over time
+fig, ax = plt.subplots()
+ax.plot(Z, states)
```

Comment:
It took me a bit of time to understand this plot; I agree once you understand it it is convincing but it perhaps warrants a slightly more extended explanation?